src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -1,38 +1,130 @@
 package org.constellation.domain.redownload
 
+import cats.effect.{ContextShift, LiftIO, Sync}
 import cats.effect.Concurrent
 import cats.effect.concurrent.Ref
 import cats.implicits._
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import org.constellation.DAO
+import org.constellation.consensus.EdgeProcessor
+import org.constellation.domain.snapshotInfo.SnapshotInfoChunk
 import org.constellation.schema.Id
+import org.constellation.storage.RecentSnapshot
+import org.constellation.util.MajorityStateChooser.NodeSnapshots
+import org.constellation.util.{HealthChecker, MajorityStateChooser}
 
-class RedownloadService[F[_]]()(implicit F: Concurrent[F]) {
+import scala.concurrent.duration._
+
+class RedownloadService[F[_]](dao: DAO)(implicit F: Concurrent[F], C: ContextShift[F]) {
```

Comment:
Please don't use `dao` anymore if possible. Current implementation of DAO is basically an antypattern and we should not add it to new services and remove from existing if possible. Instead of this, please use services explicitly. What you have in advance is a proper abstraction, so you don't need to use `LiftIO` anymore.

Also, the service lives in `domain` not the `infrastructure`, so if possible, it shouldn't know about strict implementations but abstraction.

Reply:
I mentioned this above, but RedownloadService needs readyPeers so we'll need to move peerInfo into it's own service to remove dao